src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataTypeService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.digital.ho.hocs.casework.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.dto.CaseDataType;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.exception.ApplicationExceptions;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.CASE_TYPE_LOOKUP_FAILED;
+
+@Slf4j
+@Service
+public class CaseDataTypeService {
+
+    private final InfoClient infoClient;
+
+    @Autowired
+    public CaseDataTypeService(InfoClient infoClient) {
+        this.infoClient = infoClient;
+    }
+
+    public Stream<CaseDataType> getAllCaseDataTypes(Predicate<CaseDataType> predicate) {
+        return infoClient.getAllCaseDataTypes().stream().filter(predicate);
```

Comment:
Part of me sort of feels like the Predicate stuff shouldn't be part of this method, `getAllCaseDataTypes` aludes to returning them all...

Reply:
I do however really like us using a properly cached method and pulling the value from that...